corda/corda-runtime-os
# CORE-8859 Add TLS type to the Gateway

Currently this option is mostly ignored. The gateway will still only check the server certificate, even if the TLS type is set to MUTUAL. If you change the TLS type the gateway will close all client and server connections.

Merge https://github.com/corda/corda-api/pull/757 first.

Manual Testing
===

I deployed a worker cluster using a docker image built from this PR and updated the config:
```
curl --insecure -u admin:admin  -X PUT -d @config.json  https://localhost:8888/api/v1/config
```
where config.json contains:
```
{
  "request": {
      "config": { "sslConfig": { "tlsType": "MUTUAL" } },
      "schemaVersion": {
          "major": 1,
          "minor": 0
      },
      "section": "corda.p2p.gateway",
      "version": 0
  }
}
```

I can see both the gateway client and server restart in the logs:
```
17:10:03.544 [lifecycle-coordinator-13] INFO  ReconfigurableConnectionManager-1 {} - Got new configuration
17:10:03.544 [lifecycle-coordinator-13] INFO  ReconfigurableConnectionManager-1 {} - Applying new configuration
17:10:03.544 [lifecycle-coordinator-13] INFO  net.corda.p2p.gateway.messaging.ReconfigurableConnectionManager {} - New configuration, clients for ReconfigurableConnectionManager-1 will be reconnected
17:10:03.548 [lifecycle-coordinator-14] INFO  ConnectionConfigReader-1 {} - Got new configuration
17:10:03.548 [lifecycle-coordinator-14] INFO  ConnectionConfigReader-1 {} - Applying new configuration
17:10:03.548 [lifecycle-coordinator-15] INFO  ReconfigurableHttpServer-1 {} - Got new configuration
17:10:03.548 [lifecycle-coordinator-12] INFO  ConnectionConfigReader-1 {} - Received valid config for ConnectionConfigReader-1.
17:10:03.548 [lifecycle-coordinator-15] INFO  ReconfigurableHttpServer-1 {} - Applying new configuration
17:10:03.549 [lifecycle-coordinator-15] INFO  net.corda.p2p.gateway.messaging.http.ReconfigurableHttpServer {} - New server configuration for ReconfigurableHttpServer-1 on the same port, HTTP server will have to go down
17:10:03.650 [lifecycle-coordinator-12] INFO  ReconfigurableConnectionManager-1 {} - Received valid config for ReconfigurableConnectionManager-1.
17:10:05.555 [lifecycle-coordinator-15] INFO  net.corda.p2p.gateway.messaging.http.HttpServer {} - Stopping HTTP Server 0.0.0.0:8080
17:10:05.555 [lifecycle-coordinator-15] INFO  net.corda.p2p.gateway.messaging.http.HttpServer {} - Starting HTTP Server
17:10:05.555 [lifecycle-coordinator-15] INFO  net.corda.p2p.gateway.messaging.http.HttpServer {} - Trying to bind to 0.0.0.0:8080
17:10:05.556 [lifecycle-coordinator-15] INFO  net.corda.p2p.gateway.messaging.http.HttpServer {} - Listening on port 8080
17:10:05.556 [lifecycle-coordinator-15] INFO  net.corda.p2p.gateway.messaging.http.HttpServer {} - HTTP Server started
17:10:05.556 [lifecycle-coordinator-12] INFO  ReconfigurableHttpServer-1 {} - Received valid config for ReconfigurableHttpServer-1.
```
